frontend/packages/dev-console/src/components/topology/filters/FilterDropdown.tsx

```diff
@@ -0,0 +1,69 @@
+import * as React from 'react';
+import { Select, SelectVariant, SelectOption, SelectGroup } from '@patternfly/react-core';
+import { ShowFiltersKeyValue, ExpandFiltersKeyValue, DisplayFilters } from './filter-utils';
+import './FilterDropdown.scss';
+
+type FilterDropdownProps = {
+  filters: DisplayFilters;
+  onChange: (filter: DisplayFilters) => void;
+};
+
+const FilterDropdown: React.FC<FilterDropdownProps> = ({ filters, onChange }) => {
+  const [isOpen, setIsOpen] = React.useState();
+  const [selected, setSelected] = React.useState<string[]>(
+    Object.keys(filters)
+      .filter((key) => filters[key])
+      .map((key) => ShowFiltersKeyValue[key] || ExpandFiltersKeyValue[key]),
+  );
+
+  const onToggle = (open: boolean): void => setIsOpen(open);
+  const onOptionClick = (e: React.MouseEvent, key: string, value: string): void => {
+    filters[key] = (e.target as HTMLInputElement).checked;
+    onChange(filters);
+    if (selected.includes(value)) {
+      setSelected(selected.filter((item) => item !== value));
+    } else {
+      setSelected([...selected, value]);
+    }
+  };
+
+  const showOptions = (
+    <SelectGroup label="Show">
+      {Object.keys(ShowFiltersKeyValue).map((key) => (
+        <SelectOption
+          key={key}
+          value={ShowFiltersKeyValue[key]}
+          onClick={(e: React.MouseEvent) => onOptionClick(e, key, ShowFiltersKeyValue[key])}
+        />
```

Comment:
Your `SelectOption` should use `children` as the display label and `value={key}`

Reply:
You can probably also move to using `onSelect` on the `Select` component once you update to use the key value.

Reply:
fixed in https://github.com/openshift/console/pull/3746/commits/869bfacfd4ab845e19e6951a96854344d4e874f3